### PR TITLE
API-65: add integration tests on PQB [properties product]

### PR DESCRIPTION
**Pull request by skeleton**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

add integration tests on:
- datetime
- family
- groups
- completeness
- category

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | done
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/GroupsFilterIntegration.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        parent::setUp();
+
+        if (1 === self::$count || $this->getConfiguration()->isDatabasePurgedForEachTest()) {
+            $group = $this->get('pim_catalog.factory.group')->create();
+            $this->get('pim_catalog.updater.group')->update($group, [
+                'code' => 'groupC',
+                'type' => 'RELATED'
+            ]);
+            var_dump($group);
```
Comment (pierallard) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/GroupsFilterIntegration.php]: nope ;)

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/CompletenessFilterIntegration.php [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        if (1 === self::$count || $this->getConfiguration()->isDatabasePurgedForEachTest()) {
+            $family = $this->get('pim_catalog.factory.family')->create();
+            $this->get('pim_catalog.updater.family')->update($family, [
+                'code'                   => 'familyB',
+                'attributes'             => ['sku', 'a_metric', 'a_localized_and_scopable_text_area', 'a_scopable_price'],
```
Comment (pierallard) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/CompletenessFilterIntegration.php]: I think the right way to write is "textarea" in one word

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/GroupsFilterIntegration.php (hunk 1 above)
Comment (a2xchip) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/GroupsFilterIntegration.php]: ^^ For dev purposes only

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/CompletenessFilterIntegration.php [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+
+    public function testOperatorAll()
+    {
+        $result = $this->execute([['completeness', 'ALL', 0]]);
+        $this->assert($result, ['product_one', 'product_two', 'empty_product', 'no_family']);
+    }
+
+    /**
+     * @expectedException \Pim\Component\Catalog\Exception\InvalidArgumentException
+     * @expectedExceptionMessage Attribute or field "completeness" expects an array with the key "locales" as data, "Array
```
Comment (pierallard) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/CompletenessFilterIntegration.php]: missing "

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/GroupsFilterIntegration.php (hunk 1 above)
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/GroupsFilterIntegration.php]: done ^^

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/CompletenessFilterIntegration.php (hunk 2 above)
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/CompletenessFilterIntegration.php]: maybe, but I won't change all fixtures in this PR ;)

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/CompletenessFilterIntegration.php (hunk 3 above)
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/CompletenessFilterIntegration.php]: done

**Pull request merged**